torch/csrc/utils/tensor_flatten.cpp

```diff
@@ -14,7 +14,7 @@ std::vector<TensorGroup> take_tensors(
   std::vector<TensorGroup> results;
   // an overapproximation, but at least we won't have to copy stuff around
   results.reserve(tensors.size());
-  std::map<TypeID, TensorGroup> groups;
+  std::unordered_map<DeprecatedTypeProperties*, TensorGroup> groups;
```

Comment:
This change means the order is no longer deterministic, which is what breaks the test.